include/aws/io/tls_channel_handler.h

```diff
@@ -88,6 +90,8 @@ struct aws_tls_connection_options {
 };
 
 struct aws_tls_ctx_options {
+    struct aws_allocator *allocator;
```

Comment:
If you don't want people manipulating this manually, is it worth putting it in a private header and just forward declaring it for pointers here?

Reply:
I don't, but I also don't want yet another allocation failure point, branch nonsense in the user's code. Making it private means I have to have some return a pointer to something function, which I don't really want to do.

Reply:
That's fair.